File: Java/Src/Singleton/NavigationSystem.java
package singleton;

public class NavigationSystem {
    private static volatile NavigationSystem instance;
    
    private NavigationSystem() {}

    public static NavigationSystem getInstance() {
        if (instance == null) {
            synchronized (NavigationSystem.class) {
                if (instance == null) {
                    instance = new NavigationSystem();
                }
            }
        }
        return instance;
    }

    public void navigate(String destination) {
        System.out.println("Navigating to " + destination);
    }
}

File: Python/src/singleton/navigation_system.py
class NavigationSystem:
    """Singleton Navigation System"""
    _instance = None

    def __new__(cls):
        if not cls._instance:
            cls._instance = super(NavigationSystem, cls).__new__(cls)
        return cls._instance

    def navigate(self, destination):
        print(f"Navigating to {destination}")

# Usage Example
if __name__ == "__main__":
    nav1 = NavigationSystem()
    nav2 = NavigationSystem()
    print(nav1 is nav2)  # True, single instance
    nav1.navigate("Central Park")
